backend/Controllers/UserController.cs

```diff
@@ -63,6 +70,13 @@ public async Task<IActionResult> CreateUser([FromBody] NewUserResource newUser)
             User user = new User { Mail = newUser.Mail, Password = hashedPassword, PasswordSalt = salt };
             user.PasswordSalt = salt;
 
+            string mailDomain = newUser.Mail.Substring(newUser.Mail.IndexOf('@') + 1);
+            Municipality municipality = await municipalityService.GetMunicipalityByDomainAsync(mailDomain);
+
+            if (municipality != null)
```

Comment:
Looks like a user will be automatically connected to a municipality if they have x.kommune.no or mgk.no email? Shouldnt we keep it more explicit and force the registration to also give a isMunicipality = true field in the request?

Reply:
Hmmm, could do that, but thought it may be confusing having x.kommune.no not be attached to a municipality. I am open to suggestions of a redesign if that is desired. You want a isMunicipality field to be checked also, and if that is true we try to add, and we return NOT OK if fails to add a municipality then?

Reply:
I was more thinking that if 'isMunicipality = true' or 'type = municipality' and the email was a municipality then we create a municipality user, and if not then we just create a normal user.

Reply:
I see, but if the field is true, and the mail domain does not match, that would result in a failure right?

Reply:
Yeh. Also 'type = municipality' would probably be better than 'isMunicipality = true' as its more flexible and modifiable

Reply:
Alright, will look into it later, guess I will use the UserType field. This will require some changes in the front-end though. But I guess that is not a problem.

Reply:
It's fine, I already had it made last sprint, so I'll just load up the old file.